### DISCUSSION TITLE
Hooking on scaling events

### DISCUSSION THREAD

**Discussion by or-adar** [kedacore/keda#3420] [Q&A / Need Help] [answered]
Hi,
I wanted to know if KEDA exposes some capabilities to emit events that will include information on scaling action.
It could really fit into some use cases when I'm in need of reacting to certain states:
like alerting certain services that the replica count is 0,
or alerting a service that the replica count reached the max replica count so the notified service will react accordingly.

If not, does it mean that I'll have to integrate Kubernetes events with other existing solutions like Argo Events or Robusta?
Are there any plans to support such thing?

**Comment by tomkerkhove** [comment]
Today we provide kubernetes events (see https://keda.sh/docs/2.7/operate/events/) but are closing the design spec for adding CloudEvent support (see https://github.com/kedacore/keda/issues/479)

Feel free to check the design and let us know what you think.

**Comment by or-adar** [comment, marked as answer]
Oh cool! thanks for sharing this! @tomkerkhove 
So I guess that for now, I'll have to use something like Knative or Argo Events to implement alerting of replica count,
(either way, it's going to be CloudEvent compliant, so it won't be hard to migrate when keda will emit events).

Also, I see that the design doc contains references to knative, does this mean that this feature will be based on it?

**Reply by tomkerkhove** [reply]
No, but there might be integration with it further down the line but @zroubalik can talk more about it.

Basically instead of configuring an HTTP endpoint, you can point to Knative which will receive all the events from KEDA.

**Reply by zroubalik** [reply]
Yeah, it will at act as Knative Event Source, so you can then consume those CEs in Knative.
